torch/csrc/jit/script/concrete_module_type.cpp

```diff
@@ -1,8 +1,51 @@
 #include <torch/csrc/jit/script/concrete_module_type.h>
+#include <torch/csrc/jit/pybind_utils.h>
 
 namespace torch {
 namespace jit {
 namespace script {
+
+bool isSimpleConstant(const IValue& v) {
+  return v.isBool() || v.isDouble() || v.isInt() || \
+    v.isString() || v.isNone() || v.isDevice();
+}
+
+IValue getValidConstantFromTuple(const std::string& attr_name,
```

Comment:
This entire function is copy/pasted from python without the python version removed. There should be only one source of truth for valid constants, or there will be future bugs.

Reply:
yeah python version is not used anymore, I can remove it

Reply:
actually I think I'll keep the python one. we'll just be converting from the preprocessed constant from python at this point